cmd/influxd/launcher/launcher.go

```diff
@@ -922,6 +929,18 @@ func (m *Launcher) run(ctx context.Context) (err error) {
 	return nil
 }
 
+// isAddressPortAvailable checks whether the address:port is available to listen,
+// by using net.Listen to verify that the port opens successfully, then closes the listener.
+func isAddressPortAvailable(address string, port int) (bool, error) {
+	if l, err := net.Listen("tcp", fmt.Sprintf("%s:%d", address, port)); err == nil {
+		if err := l.Close(); err != nil {
```

Comment:
Seems like this would still be racy. Two parallel test packages could contend for the same port if one package opens and closes the port, and the next package immediately opens and closes the same port.

If it's an improvement over the current code, this is probably still mergeable, but it ought to be commented that this is known to be racy.

The better option would be to instruct nats to listen on `:0` and have it report back what port it is listening on, but I assume it doesn't support the latter.

Reply:
Right, the NATS package doesn't seem to accommodate the way we are writing unit tests.

I thought the longer-term plan was to remove NATS altogether, but I can't find an open issue for that.